modules/subscription/locals.tf

```diff
@@ -0,0 +1,72 @@
+locals {
+  app_name = join(" ", ["DTS Bootstrap", join("", ["(", join(":", ["sub", azurerm_subscription.this.subscription_name]), ")"])])
+  # env      = var.value.environment == "sandbox" ? "sbox" : var.value.environment
+  groups = {
+    "Azure Kubernetes Service Cluster Admin Role" = {
+      name        = join(" ", ["DTS AKS Administrators", join("", ["(", join(":", ["sub", azurerm_subscription.this.subscription_name]), ")"])]),
+      description = "Grants aks cluster admin permissions to the ${azurerm_subscription.this.subscription_name} subscription"
+      members     = [data.azuread_group.aks_global_admin.id]
+    }
+    "Azure Kubernetes Service Cluster User Role" = {
+      name        = join(" ", ["DTS AKS Users", join("", ["(", join(":", ["sub", azurerm_subscription.this.subscription_name]), ")"])]),
+      description = "Grants aks cluster user permissions to the ${azurerm_subscription.this.subscription_name} subscription"
+      members     = []
+    }
+    "Contributor" = {
+      name        = join(" ", ["DTS Contributors", join("", ["(", join(":", ["sub", azurerm_subscription.this.subscription_name]), ")"])]),
+      description = "Grants contributor permissions to the ${azurerm_subscription.this.subscription_name} subscription"
+      members     = [var.dts_operations_group_id, azuread_service_principal.sp.object_id]
+    }
+    "Key Vault Administrator" = {
+      name        = join(" ", ["DTS Key Vault Administrators", join("", ["(", join(":", ["sub", azurerm_subscription.this.subscription_name]), ")"])]),
+      description = "Grants key vault administrator permissions to the ${azurerm_subscription.this.subscription_name} subscription"
+      members     = [var.dts_operations_group_id, azuread_service_principal.sp.object_id]
+    }
+    "Reader" = {
+      name        = join(" ", ["DTS Readers", join("", ["(", join(":", ["sub", azurerm_subscription.this.subscription_name]), ")"])]),
+      description = "Grants reader permissions to the ${azurerm_subscription.this.subscription_name} subscription"
+      members     = [data.azuread_group.ops_mgmt.object_id]
+    }
+    "Security Reader" = {
+      name        = join(" ", ["DTS Security Readers", join("", ["(", join(":", ["sub", azurerm_subscription.this.subscription_name]), ")"])]),
```

Comment:
can you simplify all this code to use interpolation it's way over complicated

e.g. 
`"DTS Security Readers (sub:${azurerm_subscription.this.subscription_name})"`